src/CSE.h

```diff
@@ -17,12 +17,17 @@ namespace Internal {
  * because exprs that come in from the front-end are small when
  * considered as a graph, but combinatorially large when considered as
  * a tree. For an example of a such a case, see
- * test/code_explosion.cpp */
-EXPORT Expr common_subexpression_elimination(Expr);
+ * test/code_explosion.cpp
+ *
+ * The last parameter determines whether all common subexpressions are
```

Comment:
Ordinarily I'd say that a custom enum is clearer for bool arguments of this type, as it makes the point-of-call easier to parse, but it's not clear to me what a terse name for the don't-lift-all case would be:

   enum CSELift { LiftAll, LiftSkipExprsThatSimplifierWouldPutBackIn };  // uh, no